README.md

```diff
@@ -9,92 +9,388 @@
+
+Let's assume for a moment that we want to be able to edit the `title` of
+individual cards. In the old days, we would simply create a listener function on
+the card and delegate to the parent component (`<Column />`) once we're done
+editing (typically on `blur`). By delegate we mean "notifying the parent" about
+our changed data (in this case the changed title).
 
-  updateState(state) {
-    this.state = state
+Now we all know how to attach change listeners by now, so we're going to skip
+this part. Let's concentrate on the remaining components instead.
 
-    this.listeners.forEach(l => l())
+The only place where we can update the state of our board / application is in
+the the `<App />` component.
+
+**Remember** Components can't ever update their `props`, they can only mutate
+their own state.
+
+Our `<Column />` component would attach an `onChangeTitle` listener to the
+`<Card />` component:
+
+```js
+class Card extends React.Component {
+  render () {
+    const {cards} = this.props;
+    return (
+      <div>
+        {cards.map(({title, id}, cardIndex) =>
+          <Card
+            key={id}
+            title={title}
+            onChangeTitle={this.handleChangeCardTitle.bind(this, cardIndex)}
+          />
+        )}
+      </div>
+    );
   }
 }
+```
+
+The `handleChangeCardTitle` itself would actually "delegate" itself to its
+parent component, in this case the actual `<Board />` component:
 
-// stores/togglerStore.js
-const Store = require('./Store')
+```js
+  handleChangeCardTitle (cardIndex, ev) {
+    this.props.onChangeCardTitle(this.props.id, cardIndex, ev)
+  }
+```
+
+The board would delegate to **its** parent, which is the actual `<App />`
+component:
+
+```js
+  handleChangeCardTitle (columnIndex, cardIndex, ev) {
+    this.props.onChangeCardTitle(columnIndex, cardIndex, ev)
+  }
+```
 
-const initialState = {
-  on: false
+The `<App />` component itself would now **actually** update the underlying
+state:
+
+```js
+class App extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = { board: [] };
+  }
+  componentDidMount () {
+    fetchBoard().then(board => this.setState({ board }));
+  }
+  // Magic comes in here:
+  handleChangeCardTitle (columnIndex, cardIndex, ev) {
+    // Actually we wouldn't even want to mutate this.state directly. Ideally we
+    // should make a copy of the respective card and apply our change there.
+    this.state.board[columnIndex][cardIndex].title = ev.target.value;
+    updateBoard(this.state).then(() => {
+      // Do some more magic here.
+    });
+  }
+  render () {
+    const {board} = this.state;
+    return <Board board={board} onChangeCardTitle={this.hanldeChangeCardTitle(ev)} />;
+  }
 }
+```
+
+### Change Listeners? — We don't need them!
+
+Now this _would_ work, but it's complicated beyond measures. It's much easier
+and less error-prone to move our state out into a separate store.
+
+```
+<App /> (can update this.state.board)
```

Comment:
Could we maybe add a leading paragraph to this overview that says something like 'as a reminder, this is what our current architecture looks like'? While reading the lesson, it felt like the overview here was going to be about the proposed solution, not about what we already have.